server/pulp/server/controllers/repository.py

```diff
@@ -154,6 +197,52 @@ def find_repo_content_units(
             yield_count += 1
 
 
+def find_units_not_downloaded(repo_id):
+    """
+    Find content units that have not been fully downloaded.
+
+    :param repo_id: ID of the repo whose units should be retrieved.
+    :type  repo_id: str
+
+    :return: The requested units.
+    :rtype:  generator
+    """
+    query_sets = get_mongoengine_unit_querysets(repo_id)
+    query_sets = [q(downloaded=False) for q in query_sets]
+    return chain(*query_sets)
+
+
+def missing_unit_count(repo_id):
+    """
+    Retrieve the number of units that have not been downloaded.
+
+    :param repo_id: ID of the repo to retrieve the missing unit count for.
+    :type  repo_id: str
+
+    :return: Number of units that have a ``downloaded`` flag set to false.
+    :rtype:  int
+    """
+    query_sets = get_mongoengine_unit_querysets(repo_id)
+    return sum([query_set(downloaded=False).count() for query_set in query_sets])
```

Comment:
Ditch the square brackets, and you'll get a generator expression, which might be slightly more efficient.